scene/gui/control.cpp

```diff
@@ -200,20 +200,41 @@ void Control::reparent(Node *p_parent, bool p_keep_global_transform) {
 void Control::get_argument_options(const StringName &p_function, int p_idx, List<String> *r_options) const {
 	ERR_READ_THREAD_GUARD;
 	Node::get_argument_options(p_function, p_idx, r_options);
+	if (!r_options->is_empty()) {
+		// If arguments exist, it means some other function was already matched.
+		return;
+	}
 
 	if (p_idx == 0) {
-		List<StringName> sn;
 		String pf = p_function;
+		Theme::DataType type = Theme::DATA_TYPE_MAX;
+
 		if (pf == "add_theme_color_override" || pf == "has_theme_color" || pf == "has_theme_color_override" || pf == "get_theme_color") {
-			ThemeDB::get_singleton()->get_default_theme()->get_color_list(get_class(), &sn);
+			type = Theme::DATA_TYPE_COLOR;
 		} else if (pf == "add_theme_style_override" || pf == "has_theme_style" || pf == "has_theme_style_override" || pf == "get_theme_style") {
-			ThemeDB::get_singleton()->get_default_theme()->get_stylebox_list(get_class(), &sn);
+			type = Theme::DATA_TYPE_STYLEBOX;
 		} else if (pf == "add_theme_font_override" || pf == "has_theme_font" || pf == "has_theme_font_override" || pf == "get_theme_font") {
-			ThemeDB::get_singleton()->get_default_theme()->get_font_list(get_class(), &sn);
+			type = Theme::DATA_TYPE_FONT;
 		} else if (pf == "add_theme_font_size_override" || pf == "has_theme_font_size" || pf == "has_theme_font_size_override" || pf == "get_theme_font_size") {
-			ThemeDB::get_singleton()->get_default_theme()->get_font_size_list(get_class(), &sn);
+			type = Theme::DATA_TYPE_FONT_SIZE;
 		} else if (pf == "add_theme_constant_override" || pf == "has_theme_constant" || pf == "has_theme_constant_override" || pf == "get_theme_constant") {
-			ThemeDB::get_singleton()->get_default_theme()->get_constant_list(get_class(), &sn);
+			type = Theme::DATA_TYPE_CONSTANT;
+		} else if (pf == "add_theme_icon_override" || pf == "has_theme_icon" || pf == "has_theme_icon_override" || pf == "get_theme_icon") {
+			type = Theme::DATA_TYPE_ICON;
+		}
```

Comment:
Would you mind reordering this in the alphabetical order? We don't do it everywhere (sometimes it's icon, style, font, color, constant), but this order here is completely arbitrary.